envisage/plugins/ipython_kernel/kernelapp.py

```diff
@@ -39,6 +39,9 @@ def atexit_unregister(func):
 else:
     from atexit import unregister as atexit_unregister
 
+# Sentinel object used to represent a missing attribute.
+_MISSING = object()
```

Comment:
any specific reason why we need to use an object instead of just using `None`?

Reply:
Yes. The values of the attributes could be `None`. There's a difference between restoring a `None` value and deleting the attribute.